populator-machinery/controller.go

```diff
@@ -108,13 +114,65 @@ type controller struct {
 	recorder             record.EventRecorder
 	referenceGrantLister referenceGrantv1beta1.ReferenceGrantLister
 	referenceGrantSynced cache.InformerSynced
+	useProviderImpl      bool
+	populate             func(context.Context, *PopulatorParams) error
+	populateComplete     func(context.Context, *PopulatorParams) (bool, error)
+}
+
+type VolumePopulatorConfig struct {
+	MasterURL     string
+	Kubeconfig    string
+	ImageName     string
```

Comment:
Which arguments are only for pod populator vs function populator? If so, then it may be better to create sub types where the user should only set one of them, like:

```
type Config struct {
  Pod *PodPopulatorConfig
  Method *MethodPopulatorConfig
}
```

Reply:
Updated